app/controllers/concerns/user/api_authentication_helper.rb

```diff
@@ -21,17 +21,37 @@ def authenticate_user!
     raise_unauthorized unless user_authenticated?
   end
 
+  def check_host
```

Comment:
The method name is confusing to me, I don't understand the difference between it and `authenticate_in_site`